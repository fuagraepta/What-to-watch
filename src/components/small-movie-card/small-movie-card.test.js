import React from 'react';
import renderer from 'react-test-renderer';
import MovieCardTemplate from './small-movie-card.jsx';

describe(`The component is rendered correctly`, () => {
  it(`Movie card correctly renders with transferred mock-movie`, () => {
    const mockMovie = {
      genre: `Comedies`,
      title: `Johnny English`,
      image: `img/johnny-english.jpg`,
      preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
    };
    const movieCard = renderer
      .create(<MovieCardTemplate
        movie = {mockMovie}
      />)
      .toJSON();

    expect(movieCard).toMatchSnapshot();
  });
});
