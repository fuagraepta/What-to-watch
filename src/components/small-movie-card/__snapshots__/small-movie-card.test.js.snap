// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The component is rendered correctly Movie card correctly renders with transferred mock-movie 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <video
      className="player__video"
      poster="img/johnny-english.jpg"
      style={
        Object {
          "height": "100%",
          "objectFit": "fill",
          "width": "100%",
        }
      }
    />
  </div>
  <h3
    className="small-movie-card__title"
    onClick={[MockFunction]}
  >
    <a
      className="small-movie-card__link"
      href="/films#d90f:fc74:a512:6497:6aab:537:251c:c35a"
    >
      Johnny English
    </a>
  </h3>
</article>
`;
